tests/ui/octal_escapes.stderr

```diff
@@ -0,0 +1,67 @@
+error: octal-looking escape in string literal
+  --> $DIR/octal_escapes.rs:4:17
+   |
+LL |     let _bad1 = "/033[0m";
+   |                 ^^^^^^^^^
+   |
+   = note: `-D clippy::octal-escapes` implied by `-D warnings`
+   = help: octal escapes are not supported, `/0` is always a null character
+help: if an octal escape is intended, use
+   |
+LL |     let _bad1 = "/x1b[0m";
+   |                 ~~~~~~~~~
+help: if the null character is intended, disambiguate using
+   |
+LL |     let _bad1 = "/x0033[0m";
+   |                 ~~~~~~~~~~~
+
+error: octal-looking escape in byte string literal
+  --> $DIR/octal_escapes.rs:5:17
+   |
+LL |     let _bad2 = b"/033[0m";
+   |                 ^^^^^^^^^^
+   |
+   = help: octal escapes are not supported, `/0` is always a null byte
+help: if an octal escape is intended, use
+   |
+LL |     let _bad2 = "/x1b[0m";
+   |                 ~~~~~~~~~
+help: if the null byte is intended, disambiguate using
+   |
+LL |     let _bad2 = "/x0033[0m";
+   |                 ~~~~~~~~~~~
+
+error: octal-looking escape in string literal
+  --> $DIR/octal_escapes.rs:6:17
+   |
+LL |     let _bad3 = "//033[0m";
+   |                 ^^^^^^^^^^^
+   |
+   = help: octal escapes are not supported, `/0` is always a null character
+help: if an octal escape is intended, use
+   |
+LL |     let _bad3 = "//x1b[0m";
+   |                 ~~~~~~~~~~~
+help: if the null character is intended, disambiguate using
+   |
+LL |     let _bad3 = "//x0033[0m";
+   |                 ~~~~~~~~~~~~~
+
+error: octal-looking escape in string literal
+  --> $DIR/octal_escapes.rs:7:17
+   |
+LL |     let _bad4 = "/01234567";
+   |                 ^^^^^^^^^^^
+   |
+   = help: octal escapes are not supported, `/0` is always a null character
+help: if an octal escape is intended, use
+   |
+LL |     let _bad4 = "/u{53977}";
+   |                 ~~~~~~~~~~~
```

Comment:
Do you know how octal sequences in C strings are specified? In a StackOverflow I found this definition:

> An octal escape sequence, which can be part of a character constant or string literal, consists of a `\` followed by exactly 1, 2, or 3 octal digits (`'0'` .. `'7'`).
>
> In `"\07777"`, the backslash is followed by 3 octal digits (0, 7, 7), which represents a character with the value `077` in octal, or `63` in decimal. In ASCII or an ASCII-derived encoding, that happens to be a question mark `'?'`.
>
> ~[StackOverflow](https://stackoverflow.com/a/11815279)

A quick test in an online editor also confirmed this. `printf("\07711")` resulted for me in an "?11". Therefore, I would think that this should only create a suggestion for `\012`. That would also simplify the lint emission code a bit :)

Reply:
You're right, it's the hexadecimal escapes that are "unlimited" in C, not the octal ones, I misremembered. Makes it easier indeed.